src/sdk/namespace_fs.js

```diff
@@ -199,6 +239,57 @@ const dir_cache = new LRUCache({
     max_usage: config.NSFS_DIR_CACHE_MAX_TOTAL_SIZE,
 });
 
+/**
+ * @typedef {{
+ *  time: number,
+ *  stat: fs.Stats,
+ *  usage: number,
+ *  sorted_entries?: fs.Dirent[],
+ * }} ReaddirVersionsCacheItem
+ * @type {LRUCache<object, string, ReaddirCacheItem>}
+ */
+const versions_dir_cache = new LRUCache({
+    name: 'nsfs-versions-dir-cache',
+    make_key: ({ dir_path }) => dir_path,
+    load: async ({ dir_path, fs_context }) => {
+        const time = Date.now();
+        const stat = await nb_native().fs.stat(fs_context, dir_path);
+        const version_path = dir_path + "/" + HIDDEN_VERSIONS_PATH;
+        let ver_dir_stat_size;
+        let is_version_path_exists = false;
+        try {
+            const ver_dir_stat = await nb_native().fs.stat(fs_context, version_path);
+            ver_dir_stat_size = ver_dir_stat.size;
+            is_version_path_exists = true;
+        } catch (err) {
+            ver_dir_stat_size = 0;
```

Comment:
What's the purpose of this error handling?

Reply:
IIUC, you want to check for the existance of `.versions`. I think you should check specifically for `ENOENT`, and not ignore other errors

Reply:
sure